### PR TITLE
Make sure that config variables can be substituted when they are substrings

### PR DESCRIPTION
**Pull request by eugmandel**

### CODE REVIEW

File: great_expectations/data_context/util.py [hunk 1]
```diff
@@ -145,22 +146,29 @@ def substitute_config_variable(template_str, config_variables_dict):
...
-    return config_variable_value
+        if config_variable_value:
+                if match.start() == 0 and match.end() == len(template_str):
+                    ret = config_variable_value
+                else:
+                    ret = template_str[:match.start()] + config_variable_value + template_str[match.end():]
+
+    return ret
+
+
```
Comment (jcampbell) [inline great_expectations/data_context/util.py:171]: Can this be simplified to:?

```suggestion
    # These two lines could even move up to substitute_all_config_variables, I think?
    vars = config_variables_dict
    vars.update(os.environ)
    return Template(template_str).safe_substitute(vars)
```

**Review by jcampbell — COMMENTED** [review]
Question first...

File: great_expectations/data_context/util.py (hunk 1 above)
Comment (Aylr) [reply great_expectations/data_context/util.py:171]: ooh that's pretty slick James.

**Review by Aylr — APPROVED** [review]
lgtm - though James' suggestion might simplify this a ton.

File: great_expectations/data_context/util.py (hunk 1 above)
Comment (eugmandel) [reply great_expectations/data_context/util.py:171]: @jcampbell @Aylr I love that this is so much shorter! Unfortunately, Template does not cover the case of the variable's value being anything other than a string. This is an issue for something like the credentials substitution with a dictionary.

**Pull request merged**